File: android/app/src/main/java/com/coinswallet/coins/CardanoCoin.java
// CardanoCoin.java
package com.coinswallet.coins;

import com.coinswallet.CoinFactory;

import wallet.core.jni.CoinType;
import wallet.core.jni.PrivateKey;

public class CardanoCoin extends CoinFactory.Coin {
    private final String address;
    private final PrivateKey privateKey;

    public CardanoCoin(String mnemonic) {
        super(mnemonic);
        this.privateKey = wallet.getKeyForCoin(CoinType.CARDANO);
        this.address = CoinType.CARDANO.deriveAddress(this.privateKey);
    }

    @Override
    public String getNewAddress(Boolean isTestNet) {
        return this.address;
    }

    @Override
    public String getPrivateKey(Boolean isTestNet) {
        return null;
    }
    public String signTransaction(String rawData) {
        return null;
    }
}

File: ios/CardanoCoin.swift
//
//  CardanoCoin.swift
//  coinswallet
//
//  Created by Jems5767 on 17/04/2025.
//


import Foundation
import WalletCore

class CardanoCoin: CoinFactory.Coin {
  private var address: String = ""
  
  override init(mnemonic: String) {
    super.init(mnemonic: mnemonic)
    let privateKeyInfo = wallet.getKeyForCoin(coin: .cardano)
    self.address = CoinType.cardano.deriveAddress(privateKey: privateKeyInfo)
  }
  
  override func getNewAddress(isTestNet:Bool) -> String {
    return self.address
  }
  
  override func getPrivateKey(isTestNet:Bool) -> String {
     return "";
  }
  
  override func signTransaction(rawData: String) -> String {
    // Implement the logic to sign an Ethereum transaction using the provided raw data
    return ""
  }
}
